src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -130,14 +129,23 @@ object WdlExpression {
             val message = failures.collect {case f: Failure[_] => f.exception.getMessage}.mkString("\n")
             Failure(new WdlExpressionException(s"Could not evaluate expression:\n$message"))
           case (successes, _) =>
-            successes.map{_.get.wdlType}.toSet match {
-              case s:Set[WdlType] if s.isEmpty =>
-                Failure(new WdlExpressionException(s"Can't have empty array declarations (can't infer type)"))
-              case s:Set[WdlType] if s.size == 1 =>
-                Success(WdlArray(WdlArrayType(s.head), successes.map{_.get}.toSeq))
-              case _ =>
-                Failure(new WdlExpressionException("Arrays must have homogeneous types"))
-            }
+            for (subtype <- WdlType.areTypesHomogeneous(successes.map(_.get)))
```

Comment:
I ended up spotting the def further down and commented on it there, but my point was  Ithink the function needs to be renamed